File: leetcode/0205-isomorphic-strings/isIsomorphic.ts
function isIsomorphic(s: string, t: string): boolean {
  let result = true;
  const s2t = new Map<string, string>();
  const t2s = new Map<string, string>();

  for (const [index, tString] of Array.from(t).entries()) {
    const sString = s[index]!;

    const t1 = s2t.get(sString);
    const s1 = t2s.get(tString);

    if ((t1 != null && t1 != tString) || (s1 != null && s1 != sString)) {
      result = false;
      break;
    }

    s2t.set(sString, tString);
    t2s.set(tString, sString);
  }

  return result;
}

export default isIsomorphic;

File: leetcode/0205-isomorphic-strings/isIsomorphic.go
package leetcode

func isIsomorphic(s string, t string) bool {
	var result = true

	var s2t = map[byte]byte{}
	var t2s = map[byte]byte{}

	for i := 0; i < len(s); i++ {
		var tString = t[i]
		var sString = s[i]

		t1, isT1Exists := s2t[sString]
		s1, isS1Exists := t2s[tString]

		if (isT1Exists && t1 != tString) || (isS1Exists && s1 != sString) {
			result = false
			break
		}

		s2t[sString] = tString
		t2s[tString] = sString
	}

	return result
}
